File: ios/MidTrim/MidTrim/Presentation/Screens/TrimDurationScreen.swift
import SwiftUI

struct TrimDurationScreen: View {
    let selectedDuration: Double
    let isPaidUser: Bool
    let onDurationSelected: (Double) -> Void
    let onCustomTap: () -> Void
    let onContinue: () -> Void
    let onBack: () -> Void

    private let durations: [Double] = [1.0, 2.0, 3.0]

    var body: some View {
        VStack(spacing: AppSpacing.lg) {
            Text("Select trim duration").font(.titleLarge).padding(.top, AppSpacing.xl)
            HStack(spacing: AppSpacing.sm) {
                ForEach(durations, id: \.self) { duration in
                    Button { onDurationSelected(duration) } label: {
                        Text("\(Int(duration))s").font(.buttonLabel).frame(maxWidth: .infinity)
                            .padding(.vertical, AppSpacing.sm)
                            .background(selectedDuration == duration ? Color.accentPrimary : Color.bgSurface)
                            .foregroundColor(selectedDuration == duration ? .textPrimary : .textSecondary)
                            .cornerRadius(AppSpacing.cornerButton)
                    }
                }
            }
            .padding(.horizontal, AppSpacing.md)
            Button(action: onCustomTap) {
                HStack {
                    Text("Custom").font(.bodyLarge)
                    if !isPaidUser { Text("🔒") }
                }
                .frame(maxWidth: .infinity).padding(AppSpacing.md)
                .background(isPaidUser ? Color.bgElevated : Color.bgSurface.opacity(0.5))
                .foregroundColor(isPaidUser ? .textPrimary : .premiumAccent)
                .cornerRadius(AppSpacing.cornerButton)
            }
            .padding(.horizontal, AppSpacing.md)
            qualityBadge
            Spacer()
            Button("Preview Trim", action: onContinue)
                .buttonStyle(.borderedProminent).tint(.accentPrimary)
                .padding(.horizontal, AppSpacing.md).padding(.bottom, AppSpacing.md)
        }
        .background(Color.bgPrimary)
        .navigationTitle("Trim Duration")
        .toolbar {
            ToolbarItem(placement: .topBarLeading) {
                Button("Back", action: onBack)
            }
        }
    }

    private var qualityBadge: some View {
        Button(action: isPaidUser ? {} : onCustomTap) {
            HStack(spacing: AppSpacing.xs) {
                Text(isPaidUser ? "Original Quality" : "720p").font(.body)
                if !isPaidUser { Text("🔒") }
            }
            .padding(.horizontal, AppSpacing.md).padding(.vertical, AppSpacing.sm)
            .background(Color.bgSurface)
            .foregroundColor(isPaidUser ? .textSecondary : .premiumAccent)
            .cornerRadius(AppSpacing.cornerButton)
        }
        .disabled(isPaidUser)
    }
}

File: android/app/src/main/java/com/izzatismail/midtrim/presentation/screens/TrimDurationScreen.kt
package com.izzatismail.midtrim.presentation.screens

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.izzatismail.midtrim.presentation.navigation.Spacing
import com.izzatismail.midtrim.ui.theme.*

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun TrimDurationScreen(
    selectedDuration: Double,
    isPaidUser: Boolean,
    onDurationSelected: (Double) -> Unit,
    onCustomTap: () -> Unit,
    onContinue: () -> Unit,
    onBack: () -> Unit,
    modifier: Modifier = Modifier
) {
    val durations = listOf(1.0, 2.0, 3.0)

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Trim Duration") },
                navigationIcon = {
                    TextButton(onClick = onBack) { Text("Back") }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.background
                )
            )
        }
    ) { padding ->
        Column(
            modifier = modifier
                .fillMaxSize()
                .padding(padding)
                .padding(horizontal = Spacing.md.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Spacer(modifier = Modifier.height(Spacing.xl.dp))

            Text(
                text = "Select trim duration",
                style = MaterialTheme.typography.headlineLarge,
                color = MaterialTheme.colorScheme.onBackground
            )

            Spacer(modifier = Modifier.height(Spacing.lg.dp))

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(Spacing.sm.dp)
            ) {
                durations.forEach { duration ->
                    FilterChip(
                        selected = selectedDuration == duration,
                        onClick = { onDurationSelected(duration) },
                        label = {
                            Text("${duration.toInt()}s")
                        },
                        modifier = Modifier.weight(1f)
                    )
                }
            }

            Spacer(modifier = Modifier.height(Spacing.sm.dp))

            Surface(
                onClick = if (isPaidUser) ({ onCustomTap() }) else onCustomTap,
                modifier = Modifier.fillMaxWidth(),
                shape = MaterialTheme.shapes.medium,
                color = if (isPaidUser)
                    MaterialTheme.colorScheme.surface
                else
                    MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.5f)
            ) {
                Row(
                    modifier = Modifier.padding(Spacing.md.dp),
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.Center
                ) {
                    Text(
                        text = if (isPaidUser) "Custom" else "Custom",
                        style = MaterialTheme.typography.bodyLarge,
                        color = if (isPaidUser) MaterialTheme.colorScheme.onSurface else PremiumAccent
                    )
                    if (!isPaidUser) {
                        Text(
                            text = " 🔒",
                            style = MaterialTheme.typography.bodyLarge,
                            color = PremiumAccent
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(Spacing.lg.dp))

            QualityBadge(isPaidUser = isPaidUser, onUpgrade = onCustomTap)

            Spacer(modifier = Modifier.weight(1f))

            Button(
                onClick = onContinue,
                modifier = Modifier.fillMaxWidth()
            ) {
                Text("Preview Trim")
            }

            Spacer(modifier = Modifier.height(Spacing.md.dp))
        }
    }
}

@Composable
private fun QualityBadge(isPaidUser: Boolean, onUpgrade: () -> Unit) {
    Surface(
        onClick = if (isPaidUser) ({}) else onUpgrade,
        shape = MaterialTheme.shapes.small,
        color = MaterialTheme.colorScheme.surfaceVariant
    ) {
        Row(
            modifier = Modifier.padding(horizontal = Spacing.md.dp, vertical = Spacing.sm.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = if (isPaidUser) "Original Quality" else "720p",
                style = MaterialTheme.typography.bodyMedium,
                color = if (isPaidUser)
                    MaterialTheme.colorScheme.onSurfaceVariant
                else
                    PremiumAccent
            )
            if (!isPaidUser) {
                Spacer(modifier = Modifier.width(Spacing.xs.dp))
                Text(
                    text = "🔒",
                    style = MaterialTheme.typography.bodyMedium,
                    color = PremiumAccent
                )
            }
        }
    }
}